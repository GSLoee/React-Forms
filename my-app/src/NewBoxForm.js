import React, {useState} from "react";


function NewBoxForm({addBox}){  
    const INITIAL_STATE = {color: "", width: "", height: ""}
    const [formData, setFormData] = useState(INITIAL_STATE)

    const handleSubmit = evt => {
        evt.preventDefault();
        addBox(formData);
        setFormData(INITIAL_STATE)
    }

    const handleChange = evt => {
        const { name, value } = evt.target
        setFormData(data => ({
            ...data,
            [name]: value
        }))
    }
    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="color">Color:</label>
            <input
            id="color"
            name="color"
            value={formData.color}
            onChange={handleChange}
            />

            <label htmlFor="width">Width:</label>
            <input
            id="width"
            name="width"
            value={formData.width}
            onChange={handleChange}
            />

            <label htmlFor="height">Height:</label>
            <input
            id="height"
            name="height"
            value={formData.height}
            onChange={handleChange}
            />

            <button>Make Box</button>
        </form>
    )
}

export default NewBoxForm;