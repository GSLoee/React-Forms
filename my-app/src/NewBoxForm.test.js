import React from "react";
import {render, fireEvent} from '@testing-library/react'
import NewBoxForm from "./NewBoxForm";

it("render the form", function() {
    render(<NewBoxForm />)
})

it('matches the snapshot', function(){
    const { asFragment } = render(<NewBoxForm />);
    expect(asFragment()).toMatchSnapshot()
})