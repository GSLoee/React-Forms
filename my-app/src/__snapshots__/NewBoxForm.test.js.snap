// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="color"
    >
      Color:
    </label>
    <input
      id="color"
      name="color"
      value=""
    />
    <label
      for="width"
    >
      Width:
    </label>
    <input
      id="width"
      name="width"
      value=""
    />
    <label
      for="height"
    >
      Height:
    </label>
    <input
      id="height"
      name="height"
      value=""
    />
    <button>
      Make Box
    </button>
  </form>
</DocumentFragment>
`;
